File: backend/app/api/categories.py
"""
Categories API endpoints
"""

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from typing import List

from app.core.database import get_db
from app.models.category import Category
from app.schemas.category import CategoryResponse

router = APIRouter(prefix="/categories", tags=["categories"])


@router.get("/", response_model=List[CategoryResponse])
def get_categories(
    is_active: bool = None,
    db: Session = Depends(get_db)
):
    """
    Get all categories, optionally filtered by active status
    """
    query = db.query(Category)

    if is_active is not None:
        query = query.filter(Category.is_active == is_active)

    categories = query.order_by(Category.priority.desc()).all()
    return categories


@router.get("/{category_id}", response_model=CategoryResponse)
def get_category(category_id: int, db: Session = Depends(get_db)):
    """
    Get a specific category by ID
    """
    category = db.query(Category).filter(Category.id == category_id).first()
    if not category:
        from fastapi import HTTPException
        raise HTTPException(status_code=404, detail="Category not found")
    return category

File: frontend/src/services/categories.ts
import api from './api';
import type { Category } from '@/types';

export const categoriesService = {
  // Get all categories
  getCategories: async () => {
    const response = await api.get<Category[]>('/categories');
    return response.data;
  },

  // Get active categories only
  getActiveCategories: async () => {
    const response = await api.get<Category[]>('/categories?is_active=true');
    return response.data;
  },
};

export default categoriesService;
